### PR TITLE
feat: add status neighborhoods endpoint

### PR DESCRIPTION
**Pull request by martinconic**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [x] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
Part 7 of [#4842](https://github.com/ethersphere/bee/issues/4842)

### CODE REVIEW

File: openapi/SwarmCommon.yaml [hunk 1]
```diff
@@ -933,6 +933,23 @@ components:
           items:
             $ref: "#/components/schemas/StatusSnapshotResponse"
 
+    StatusNeighborhoodResponse:
+      type: object
+      properties:
+        address:
+          type: string
+        chunkCount:
```
Comment (ldeffenb) [inline openapi/SwarmCommon.yaml:941]: Is this `reserveSize` or `reserveSizeWithinRadius` or something completely different?

Comment (istae) [reply openapi/SwarmCommon.yaml:941]: This field should be called `reserveSizeWithinRadius`. Good call.

Comment (martinconic) [reply openapi/SwarmCommon.yaml]: renamed

File: openapi/SwarmCommon.yaml [hunk 2]
```diff
@@ -933,6 +933,23 @@ components:
           items:
             $ref: "#/components/schemas/StatusSnapshotResponse"
 
+    StatusNeighborhoodResponse:
+      type: object
+      properties:
+        address:
+          type: string
```
Comment (janos) [inline openapi/SwarmCommon.yaml:940]: Maybe to be more specific as it is a string of a number in a binary representation?

File: pkg/storer/reserve.go [hunk 3]
```diff
@@ -498,8 +498,8 @@ func (db *DB) SubscribeBin(ctx context.Context, bin uint8, start uint64) (<-chan
 }
 
 type NeighborhoodStat struct {
-	Address    swarm.Address
-	ChunkCount int
+	Address                 swarm.Address
```
Comment (janos) [inline pkg/storer/reserve.go:501]: This field suggests that it is a swarm address, but it looks to me that it represents a neighborhood just using the swarm.Address type. I am questioning if both the field name and its type are correct. Maybe the field name can stay the same if the type would be of the specific neighborhood type. It looks to me that neighborhood would benefit from its own type, for example swarm.Neighborhood instead to use swarm.Address as a container that may mislead the code reader. There are technical similarities between two types but the semantics of them is different. Another option, field name could be Neighborhood for clarity.

File: pkg/api/status.go [hunk 4]
```diff
@@ -159,3 +169,42 @@ func (s *Service) statusGetPeersHandler(w http.ResponseWriter, r *http.Request)
...
+
+	if len(nhoods) == 0 {
+		jsonhttp.NotFound(w, "neighborhoods not found")
+		return
+	}
+
+	for _, n := range nhoods {
+		binaryAddr := ""
+		for _, b := range n.Address.Bytes() {
+			binaryAddr += fmt.Sprintf("%08b ", b)
```
Comment (janos) [inline pkg/api/status.go:201]: I think that the binary representation could a method of a potentially new type swarm.Neighborhood. Just a suggestion.

Comment (martinconic) [reply pkg/api/status.go:201]: Searched for such a thing but could not find it, `bitStr(` not the best naming

**Review by istae — APPROVED** [review]

**Pull request merged**